libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteStepProgressTest.java

```diff
@@ -29,83 +28,60 @@
-    DirectionsRoute route = response.routes().get(0);
-
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(0)
-      .legDistanceRemaining(0)
-      .distanceRemaining(0)
-      .directionsRoute(route)
-      .stepIndex(route.legs().get(0).steps().size() - 1)
-      .legIndex(0)
-      .build();
+  public void stepDistance_equalsZeroOnOneCoordSteps() throws Exception {
+    DirectionsRoute route = loadChipotleTestRoute();
+    int stepIndex = route.legs().get(0).steps().size() - 1;
+    RouteProgress routeProgress = buildTestRouteProgress(route, 0, 0,
+      0, stepIndex, 0);
     RouteStepProgress routeStepProgress = routeProgress.currentLegProgress().currentStepProgress();
 
-    assertNotNull("should not be null", routeStepProgress);
+    assertNotNull(routeStepProgress);
     assertEquals(1, routeStepProgress.fractionTraveled(), DELTA);
     assertEquals(0, routeStepProgress.distanceRemaining(), DELTA);
     assertEquals(0, routeStepProgress.distanceTraveled(), DELTA);
     assertEquals(0, routeStepProgress.durationRemaining(), DELTA);
   }
 
   @Test
-  public void distanceRemaining_equalsStepDistanceAtBeginning() {
+  public void distanceRemaining_equalsStepDistanceAtBeginning() throws Exception {
+    DirectionsRoute route = buildTestDirectionsRoute();
+    RouteLeg firstLeg = route.legs().get(0);
+
     LineString lineString
       = LineString.fromPolyline(firstLeg.steps().get(5).geometry(), Constants.PRECISION_6);
     double stepDistance = TurfMeasurement.length(lineString, TurfConstants.UNIT_METERS);
 
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(firstLeg.steps().get(5).distance())
-      .legDistanceRemaining(firstLeg.distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(4)
-      .legIndex(0)
-      .build();
+    double stepDistanceRemaining = firstLeg.steps().get(5).distance();
+    double legDistanceRemaining = firstLeg.distance();
+    double distanceRemaining = route.distance();
+    int stepIndex = 4;
+    RouteProgress routeProgress = buildTestRouteProgress(route, stepDistanceRemaining,
+      legDistanceRemaining, distanceRemaining, stepIndex, 0);
     RouteStepProgress routeStepProgress = routeProgress.currentLegProgress().currentStepProgress();
 
     assertEquals(stepDistance, routeStepProgress.distanceRemaining(), BaseTest.LARGE_DELTA);
   }
 
   @Test
-  public void distanceRemaining_equalsCorrectValueAtIntervals() {
+  public void distanceRemaining_equalsCorrectValueAtIntervals() throws Exception {
+    DirectionsRoute route = buildTestDirectionsRoute();
+    RouteLeg firstLeg = route.legs().get(0);
+    LegStep firstStep = route.legs().get(0).steps().get(0);
+
     LineString lineString
       = LineString.fromPolyline(firstStep.geometry(), Constants.PRECISION_6);
     double stepDistance = TurfMeasurement.length(lineString, TurfConstants.UNIT_METERS);
```

Comment:
What about extracting `// Chop the line in small pieces` block of code into a private method and give it a name based on the comment? This way the test will be easier to read and understand and comment will become superfluous so it won't be necessary. Same thing with `double stepSegments = 5000; // meters` 👇

Also 👀 a bunch of `// Chop the line in small pieces` blocks of code in other tests, if it's duplicated code we should consolidate it and use it across the tests.